### DISCUSSION TITLE
Ask for an example for `SNAPEdgeListPartitionReader`

### DISCUSSION THREAD

**Discussion by nie93** [networkit/networkit#701] [Q&A]
I downloaded the dataset from Stanford SNAP network repository. And I am trying to use `SNAPEdgeListPartitionReader` to read the `*.edges` files in the dataset. But I cannot understand the `mapNodeIds` of the three input arguments for `read(std::string path, std::unordered_map<node,node>& mapNodeIds, Graph& G)`. 

Would anybody help me with an example using `SNAPEdgeListPartitionReader.read()`?

**Comment by hmeyerhenke** [comment]
Don't you want to read the graphs only? This can be done with the read method, see the user guide. Probably something like this:

G = readGraph("your.graph", Format.SNAP)

**Comment by hmeyerhenke** [comment]
Some context: SNAPEdgeListPartitionReader also reads a partition of the vertex set, I believe.

**Comment by avdgrinten** [comment]
If I understand correctly, SNAPEdgeListPartitionReader needs the input node ID -> internal node ID map that SNAPGraphReader emits. However, I'm not too familiar with the SNAP format and I have not really looked at the code. Can someone confirm this?

**Comment by avdgrinten** [comment]
@nie93 Did that solve your problem?

**Comment by nie93** [comment]
@avdgrinten Not really..

**Comment by nie93** [comment]
If anyone can help how to formulate the `mapNodeIds` in ` networkit/networkit/cpp/io/SNAPEdgeListPartitionReader.cpp` using SNAP format, it would be great.

An example of SNAPEdgeListPartition can refer to `http://snap.stanford.edu/data/facebook.tar.gz`.

Thanks!

**Comment by avdgrinten** [comment]
Did you try passing the map (it's a public member variable) from SNAPGraphReader to SNAPEdgeListPartitionReader?

**Comment by nie93** [comment]
SNAP dataset format comprises of 5 data formats for each partition, which are `*.circles`, `*.edges`, `*.egofeat`, `*.feat`, `*.featnames`. Based on my use case, I only need to use `.edges` files. 

But what I don't get is, which kind of input data is designed for `SNAPGraphReader`. It returns `RuntimeError: unable to read from file` when I tried to read `.edges` files using it.
